shared/src/main/scala/async/Async.scala

```diff
@@ -127,6 +131,30 @@ object Async:
 
   end OriginalSource
 
+  object Source:
+    /** Create a [[Source]] containing the given values, resolved once for each. */
+    def values[T](values: T*) =
+      import scala.collection.JavaConverters._
+      val q = java.util.concurrent.ConcurrentLinkedQueue[T](values.asJavaCollection)
+      new Source[T]:
+        override def poll(k: Listener[T]): Boolean =
+          if q.isEmpty() then return false
```

Comment:
I am curious: does the `return` give some running time improvement?

Reply:
Nice catch! In fact no, it doesnt and may slow it down :P I'll remove it, thanks!